network/dag/bbolt_tree.go

```diff
@@ -82,7 +82,7 @@ func (store *bboltTree) isEmpty() bool {
 
 // dagCallback inserts a transaction reference to the in-memory tree and to persistent storage.
 // The tree is not aware of previously seen transactions, so it should be transactional with updates to the dag.
-func (store *bboltTree) dagObserver(ctx context.Context, transaction Transaction, _ []byte) {
+func (store *bboltTree) dagObserver(ctx context.Context, transaction Transaction, _ []byte) error {
```

Comment:
why this change?

Reply:
This is due to nats events being added as observer. The events will be the main mechanism of transfering TX content between engines. This is an operation that MUST succeed when a bbolt transaction commits. Therefore the event is published from within the bbolt transaction. If the event publishing fails, the transactino fails.